python-lib/dku_kube/autoscaler.py

```diff
@@ -1,26 +1,46 @@
-import os, sys, json, yaml, logging, random, time
+import os, json, logging
 from .kubectl_command import run_with_timeout
+from dku_utils.access import _is_none_or_blank
+from dku_utils.tools_version import strip_kubernetes_version
+
+AUTOSCALER_IMAGES = {
+    '1.24': 'v1.24.3',
+    '1.25': 'v1.25.3',
+    '1.26': 'v1.26.4',
+    '1.27': 'v1.27.3',
+    '1.28': 'v1.28.0'
+}
 
 def has_autoscaler(kube_config_path):
     env = os.environ.copy()
     env['KUBECONFIG'] = kube_config_path
     cmd = ['kubectl', 'get', 'pods', '--namespace', 'kube-system', '-l', 'app=cluster-autoscaler', '--ignore-not-found']
-    logging.info("Create autoscaler with : %s" % json.dumps(cmd))
+    logging.info("Checking autoscaler presence with : %s" % json.dumps(cmd))
     out, err = run_with_timeout(cmd, env=env, timeout=5)
     return len(out.strip()) > 0
 
-def add_autoscaler_if_needed(cluster_id, kube_config_path):
+def add_autoscaler_if_needed(cluster_id, cluster_config, cluster_def, kube_config_path):
     if not has_autoscaler(kube_config_path):
+        print("Amandine K8S version: %s" % cluster_config.get("k8sVersion", None))
+        kubernetes_version = cluster_config.get("k8sVersion", None)
+        if _is_none_or_blank(kubernetes_version):
+            kubernetes_version = cluster_def.get("Version")
+        kubernetes_version = strip_kubernetes_version(kubernetes_version)
```

Comment:
Now that we don't make an extra call, we could always just take the `cluster_def.get("Version")` version, but I think it's fine to leave it like this as well.